documentation/external_ssl_termination.markdown

```diff
@@ -5,21 +5,61 @@ canonical: "/puppetserver/latest/external_ssl_termination.html"
+In network configurations that require external SSL termination, there are some important differences between configuring the Apache/Passenger stack and configuring Puppet Server. Use the following steps to configure external SSL termination.
 
-Delivery of the SSL certificate to Puppet is the biggest difference between Ruby Puppet and Puppet Server. Under Apache/Passenger, the client's certificate is delivered via an environment variable passed between Apache and the Ruby Puppet process. This is not an option in Puppet Server; instead, the certificate must be passed to Puppet Server via an HTTP header. The header is `X-Client-Cert` and must contain the client's PEM-formatted (Base-64) certificate in a single URI-encoded string. Note that URL encoding is not sufficient; all space characters must be encoded as `%20` and not `+` characters.
-
-> **Note**: Ruby Puppet master code uses the `ssl_client_header` and `ssl_client_verify_header` values to authorize the request per the rules in `auth.conf`. The `X-Client-Cert` value is only used for distilling trusted facts from extensions on the client certificate. To minimize the size of the request payload, you might want to omit the `X-Client-Cert` header from the request if trusted facts are not being used.
+## Disable HTTPS for Puppet Server
 
 You'll need to turn off SSL and have Puppet Server use the HTTP protocol instead: remove the `ssl-port` and `ssl-host` settings from the `conf.d/webserver.conf` file and replace them with `port` and `host` settings. See [Configuring the Webserver Service](https://github.com/puppetlabs/trapperkeeper-webserver-jetty9/blob/master/doc/jetty-config.md) for more information on configuring the web server service.
 
-To allow Puppet Server to recognize the HTTP headers described, open `config.d/master.conf` and add  `allow-header-cert-info: true` to the `webserver` config block. See [Puppet Server Configuration](./configuration.markdown) for more information on the `master.conf` file. Without `allow-header-cert-info` set to 'true', none of these HTTP headers are recognized by Puppet Server.
+## Allow Client Cert Data From HTTP Headers
+
+When using external SSL termination, Puppet Server expects to receive client certificate information via some HTTP headers.
 
-You'll need to reboot Puppet Server for the changes to take effect. The `ssl_client_header` and `ssl_client_verify_header` options in the `puppet.conf` file will then be enabled and work exactly as documented in Ruby Puppet. Please see the [Puppet Documentation](https://docs.puppetlabs.com/references/3.7.latest/configuration.html#sslclientheader) for more detail on these.
+By default, reading this data from headers is disabled. To allow Puppet Server to recognize it, open `config.d/master.conf` and add  `allow-header-cert-info: true` to the `webserver` config block. See [Puppet Server Configuration](./configuration.markdown) for more information on the `master.conf` file.
```

Comment:
It may be worth noting that the `master.conf` file isn't actually created at installation time for either for FOSS or PE for any current releases so the user would need to create the entire file at this point.  The config block section under which the `allow-header-cert-info` setting resides is named "master", not "webserver".  If no "master" config block and/or `allow-header-cert-info` setting is present in the configuration, Puppet Server will use a default value of "false" for `allow-header-cert-info`.